flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointRequestDecider.java

```diff
@@ -136,6 +136,11 @@
 				.map(unused -> queuedRequests.pollFirst());
 		}
 
+		CheckpointTriggerRequest first = queuedRequests.first();
+		if (first.isForce() || !first.isPeriodic) {
```

Comment:
Can we make the connection between this check and `onTooEarly` more explicit?
For example, by inverting this condition and putting `onTooEarly` inside.

Reply:
Hi, @rkhachatryan, thanks for your proposal. As you say, I will make the connection more explicit by inverting the condition and putting onTooEarly inside.

Reply:
Hi, @DG-Wangtao, thanks!